app/scripts/controllers/metametrics-controller.ts

```diff
@@ -188,28 +256,91 @@ export type MetaMetricsControllerState = {
   >;
 };
 
+/**
+ * Returns the state of the {@link MetaMetricsController}.
+ */
+export type MetaMetricsControllerGetStateAction = ControllerGetStateAction<
+  typeof controllerName,
+  MetaMetricsControllerState
+>;
+
+/**
+ * Actions exposed by the {@link MetaMetricsController}.
+ */
+export type MetaMetricsControllerActions = MetaMetricsControllerGetStateAction;
+
+/**
+ * Event emitted when the state of the {@link MetaMetricsController} changes.
+ */
+export type MetaMetricsControllerStateChangeEvent = ControllerStateChangeEvent<
+  typeof controllerName,
+  MetaMetricsControllerState
+>;
+
+export type MetaMetricsControllerEvents = MetaMetricsControllerStateChangeEvent;
+
+/**
+ * Actions that this controller is allowed to call.
+ */
+export type AllowedActions =
+  | PreferencesControllerGetStateAction
+  | NetworkControllerGetStateAction
+  | NetworkControllerGetNetworkClientByIdAction;
+
+/**
+ * Events that this controller is allowed to subscribe.
+ */
+export type AllowedEvents =
+  | PreferencesControllerStateChangeEvent
+  | NetworkControllerNetworkDidChangeEvent;
+
+/**
+ * Messenger type for the {@link MetaMetricsController}.
+ */
+export type MetaMetricsControllerMessenger = RestrictedControllerMessenger<
+  typeof controllerName,
+  MetaMetricsControllerActions | AllowedActions,
+  MetaMetricsControllerEvents | AllowedEvents,
+  AllowedActions['type'],
+  AllowedEvents['type']
+>;
+
 type CaptureException =
   | typeof sentryCaptureException
   | ((err: unknown) => void);
 
 export type MetaMetricsControllerOptions = {
-  initState: Partial<MetaMetricsControllerState>;
+  state: Partial<MetaMetricsControllerState>;
```

Comment:
According to the [controller guidelines](https://github.com/MetaMask/core/blob/main/docs/writing-controllers.md#accept-an-optional-partial-representation-of-state) `state` should be optional, so this should be:

``` suggestion
  state?: Partial<MetaMetricsControllerState>;
```

This is consistent with how `initState` was being treated below, I think.

Reply:
I believe we should add this to the controller guidelines too as we default the state to {} but we didn't specify that state should be optional

Reply:
Oh good point, that's not clear. I'll make a ticket for that.

Reply:
It’s clear that by default, a property should be considered optional if it’s assigned a default value. However, it would be worthwhile to update the type definition accordingly https://github.com/MetaMask/core/blob/main/docs/writing-controllers.md?plain=1#L52.